From 8cf1faf49b0c0d6b1868f3d806eb5b5cead7c32b Mon Sep 17 00:00:00 2001
From: Priyanka Chikane <priyanka.chikane@intel.com>
Date: Tue, 7 Jun 2022 12:38:13 +0000
Subject: [PATCH] Allow checking of U-Boot shell, before FFUJ check

Uboot shell login is checked before the FFUJ jumper check, so now the
user can enter into uboot command line interface on hitting any key.

Tested:
-With FFUJ enabled and autoboot stopped, output is:
Hit any key to stop autoboot:  0
ast#
-With FFUJ enabled and autoboot continued, output is:
Hit any key to stop autoboot:  0
FFUJ: 1
ast#

Signed-off-by: Priyanka Chikane <priyanka.chikane@intel.com>
---
 common/autoboot.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/common/autoboot.c b/common/autoboot.c
index 8a99780423..e23de568fe 100644
--- a/common/autoboot.c
+++ b/common/autoboot.c
@@ -271,13 +271,12 @@ static int abortboot(int bootdelay)
 {
 	int abort = 0;
 
-	abort =	gpio_abort();
-	if (abort)
-		goto exit;
-
 	if (bootdelay >= 0)
 		abort = __abortboot(bootdelay);
 
+	if (!abort)
+		abort =	gpio_abort();
+
 #ifdef CONFIG_SILENT_CONSOLE
 	if (abort)
 		gd->flags &= ~GD_FLG_SILENT;
-- 
2.25.1

